src/Field/Content.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace Message\Cog\Field;
+
+use Message\Cog\Validation\Validator;
+
+/**
+ * Basic content class.
+ *
+ * @author Sam Trangmar-Keates <sam@message.co.uk>
+ */
+class Content implements ContentInterface
+{
+	protected $_fields = array();
+	protected $_validator;
+
+	/**
+	 * Set a content part.
+	 *
+	 * @param string                             $var   Content part name
+	 * @param FieldInterface|RepeatableContainer $value The content part
+	 *
+	 * @throws \InvalidArgumentException If the content part was not a valid instance
+	 */
+	public function __set($var, $value)
+	{
+		if (!($value instanceof FieldInterface || $value instanceof RepeatableContainer)) {
+			throw new \InvalidArgumentException(sprintf(
+				'Page content must be a `FieldInterface` or a `RepeatableContainer`, `%s` given',
+				get_class($value)
+			));
+		}
+
+		$this->_fields[$var] = $value;
+	}
+
+	/**
+	 * Get a content part by name.
+	 *
+	 * @param  string $var Content part name
+	 *
+	 * @return FieldInterface|RepeatableContainer $value The content part
+	 */
+	public function __get($var)
+	{
+		return array_key_exists($var, $this->_fields) ? $this->_fields[$var] : null;
+	}
+
+	/**
+	 * Check if a content part is set on this object.
+	 *
+	 * @param  string  $var Content part name
+	 *
+	 * @return boolean
+	 */
+	public function __isset($var)
+	{
+		return isset($this->_fields[$var]);
+	}
+
+	public function set($key, $value)
+	{
+		return $this->__set($key, $value);
```

Comment:
Swap these methods with the magic methods, i.e. __set() should call set() not the other way around. Also add docblocks